### DISCUSSION TITLE
Salvaging Data from Dead Sever

### DISCUSSION THREAD

**Discussion by Kriegbaum** [inventree/InvenTree#8782] [Q&A] [answered]
Hello!
I recently had a server die and I'm attempting to get a new inventree instance up and running with my inventory and I am now seeing that I've made some mistakes in my backup procedure.

I possess a copy of the inventree data directory and a backup archive from the old server, and I'm seeing that I should be running the same version of Inventree on the same setup if I want to restore successfully. The problem is that I don't actually know what I was running before, I did not take note of it as I didn't think it was critical.

Is there enough info sitting in the inventree data directory to successfully recreate my old setup?
When I do routine backups should I just be exporting the data for migration every time so that I don't wind up in this mess again?
Are there any tricks that can be used to save myself some pain here?

Any guidance would be appreciated, reconstructing the database will take quite a while if that's what is necessary

**Comment by Kriegbaum** [comment]
If there is even a way to manually view the database (secret key has been preserved in the backup) I could use that to rebuild the database faster than taking stock of my supplies

**Comment by SchrodingersGat** [comment]
If the database is still available (and not corrupted) then you should be able to make use of it. If you do not have backups to restore from, the general idea would be:

1. Setup a new InvenTree installation
2. Replace the "new" database with your "old" one
3. Run `invoke update` to bring the old database "up to date"
4. Profit

Point `2.` depends on what database backend you have been using, and your original installation / setup method (bare metal? docker? etc?)

So:

a) What installation method did you use originally
b) What database backend were you originally using?

**Comment by Kriegbaum** [comment, marked as answer]
This experience has definitely taught me that I need to be saving some notes with my backups :laughing: 
The data folder I backed up was a mounted docker volume, and inside it I have a `pgdb` directory, there are a number of postgresql configuration files in there an a `PG_VERSION` file that says 13, I think its safe to assume it was running postgresql
I also do have backups in the backup directory. The most recent one is only a week old so I can do that if it is usable.

I followed a guide that had this set up through docker compose, pretty sure it was the one you provided in the docs.

I was reading in other posts on this forum that if you restore a backup, it needs to be the same inventree version that the backup came from. I'm not finding anything in the data folder that hints at what version I was running though. 

I'll try grabbing a version from a year ago (around the time I spun up the server in question) and see if I can restore it to that

**Reply by SchrodingersGat** [reply]
You *should* be able to setup a new docker install, and then delete `./data/pgdb/` and replace it with your salvaged contents.

Then, to bring the database up to date with the code, simply run `docker compose run --rm inventree-server invoke update`

**Reply by Kriegbaum** [reply]
Success! I appreciate you jumping in on this so quickly, I was not looking forward to rebuilding everything. 
I've learned a few things about good practices (I'm more of a hardware person so I'm kinda feeling my way through some of this server admin stuff)
Got me a revision controlled docker compose/caddyfile/env file and an ansible play that will get the server up from a bare debian install in a few minutes, I think I'm going to add a data export step to JSON to my daily backup procedure since it looks like it doesn't take all that long and it gives me more options for recovery later.

**Reply by SchrodingersGat** [reply]
@Kriegbaum glad to hear it worked out for you :)
